package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -388,79 +316,147 @@
    :inherited-members:
 
 """
-from MDAnalysis.lib.correlations import autocorrelation, correct_intermittency
-import MDAnalysis.analysis.hbonds
 from itertools import zip_longest
 import logging
-import warnings
+
 import numpy as np
 
+from MDAnalysis.lib.log import ProgressBar
+from MDAnalysis.analysis.base import AnalysisBase
+from MDAnalysis.lib.correlations import autocorrelation, correct_intermittency
 
 logger = logging.getLogger('MDAnalysis.analysis.waterdynamics')
-from MDAnalysis.lib.log import ProgressBar
 
 
-class WaterOrientationalRelaxation(object):
+class WaterOrientationalRelaxation(AnalysisBase):
     r"""Water orientation relaxation analysis
 
-    Function to evaluate the Water Orientational Relaxation proposed by Yu-ling
-    Yeh and Chung-Yuan Mou [Yeh1999_]. WaterOrientationalRelaxation indicates
-    "how fast" water molecules are rotating or changing direction. This is a
-    time correlation function given by:
+    Class evaluating the water orientational relaxation proposed by Yu-ling
+    Yeh and Chung-Yuan Mou [Yeh1999_].
+    `WaterOrientationalRelaxation` (WOR) indicates
+    "how fast" water molecules are rotating or changing direction.
+    If WOR is very stable we can assume that water molecules are
+    rotating/changing direction very slowly, on the
+    other hand, if WOR decay very fast, we can assume that water molecules are
+    rotating/changing direction very fast. The results are
+    time correlation functions given by:
 
     .. math::
         C_{\hat u}(\tau)=\langle \mathit{P}_2[\mathbf{\hat{u}}(t_0)\cdot\mathbf{\hat{u}}(t_0+\tau)]\rangle
 
-    where :math:`P_2=(3x^2-1)/2` is the second-order Legendre polynomial and :math:`\hat{u}` is
-    a unit vector along HH, OH or dipole vector.
-
+    where :math:`P_2=(3x^2-1)/2` is the second-order Legendre polynomial
+    and :math:`\hat{u}` is a unit vector along HH, OH or dipole vector.
```

Comment:
Please add a warning that this class assumes a 3-point water model where each water residue contains O as the first entry and hydrogen 1 and 2 as second and third entry.

This warning should have been here from the beginning. Even better would be a more robust approach to choosing O and H1, H2 but that's not the job of this PR.